src/commands/join.cpp

```diff
@@ -18,6 +18,8 @@ int		command_join(Message &message, Dependencies &deps)
 	std::vector<std::string>::const_iterator	key_it = keys.begin();
 	while (chan_it != channels.end())
 	{
+		if (!_base_channel::isValidName(*chan_it))
+			return 1;
```

Comment:
Should send an `403 ERR_NOSUCHCHANNEL` and skip this iteration (continue won't work as the incrementation happens at the end).